src/main/java/org/openstreetmap/atlas/geography/boundary/CountryBoundaryMap.java

```diff
@@ -1118,6 +1121,55 @@ public void initializeGridIndex(final Set<String> countries)
         }
     }
 
+    /**
+     * Filters the given iterable of {@link ComplexBoundary}'s to only allow a single country to
+     * claim an outer boundary polygon.
+     *
+     * @param complexBoundaries
+     *            The {@link ComplexBoundary}'s to filter
+     * @return An iterable of {@link ComplexBoundary} with no duplicate outer polygons
+     */
+    public Iterable<ComplexBoundary> resolveOverlappingBorders(
+            final Iterable<ComplexBoundary> complexBoundaries)
+    {
+        final List<ComplexBoundary> dedupedComplexBoundaries = new ArrayList<>();
+        final Set<org.openstreetmap.atlas.geography.Polygon> processedOuters = new HashSet<>();
+        final List<ComplexBoundary> complexBoundariesList = Lists.newArrayList(complexBoundaries);
+        Collections.sort(complexBoundariesList, (firstBoundary, secondBoundary) ->
```

Comment:
Wouldn't it be better to use a SortedSet here? That way it would be sorted at insertion time. Can this contain duplicates? If yes, then Set won't work.

Reply:
Since we pass in an `Iterable`, we would still need to convert from an `Iterable` to a `Collection` before being able to insert into a `SortedSet`, right? Therefore we'd still need to an insertion prior to the sorting.

Also, to my mind this list could contain duplicates, as this is the list of `ComplexBoundary` objects, not the set of outer polygons.